icx/src/main.rs

```diff
@@ -324,13 +324,7 @@ pub fn get_effective_canister_id(
                 Ok(in_args.target_canister)
             }
             MgmtMethod::BitcoinGetBalanceQuery | MgmtMethod::BitcoinGetUtxosQuery => {
-                #[derive(CandidType, Deserialize)]
-                struct In {
-                    network: BitcoinNetwork,
-                }
-                let in_args = Decode!(arg_value, In)
-                    .with_context(|| format!("Argument is not valid for {method_name}"))?;
-                Ok(in_args.network.effective_canister_id())
+                Ok(Principal::management_canister())
```

Comment:
This is a very surprising change. Does it work on the main net? I believe it can work locally, but the main net won't know how to route `aaaaa-aa`.

Reply:
It did work locally as verified in the sdk PR.

And the [spec](https://internetcomputer.org/docs/current/references/ic-interface-spec/#http-effective-canister-id) says:
> If the request is an update call to the Management Canister (aaaaa-aa), then:
>   * If the call is to the provisional_create_canister_with_cycles method, then any principal can be used as the effective canister id for this call.
>   * Otherwise, if the arg is a Candid-encoded record with a canister_id field of type principal, then the effective canister id must be that principal.
>   * Otherwise, the call is rejected by the system independently of the effective canister id.
> 
> If the request is an update call to a canister that is not the Management Canister (aaaaa-aa) or if the request is a query call, then the effective canister id must be the canister_id in the request.

In this case, the final rule should be in effect since it is a query call and the effective_canister_id must be the canister_id in the request which is `aaaaa-aa`.

However, with a quick experiment, it seems that the query call doesn't work on the mainnet.
```
❯ dfx canister call --query aaaaa-aa bitcoin_get_balance_query '(record { network = variant { mainnet }; address = "bc1q7s4xl56gy5q5d5mwemv0nn6mv99tmx6x0xvdae"})' --network=ic
WARN: Cannot fetch Candid interface for bitcoin_get_balance_query, sending arguments with inferred types.
Error: Failed query call.
Caused by: Failed query call.
  The replica returned an HTTP Error: Http Error: status 400 Bad Request, content type "text/plain; charset=utf-8", content: subnet_not_found
```
The dfx being used is the current master branch with my fix.

I would interpret this as a wrong behavior of the replica  implementation.

I'll ask the cross-chain team.

Reply:
The spec is wrong. Effective canister id can never be `aaaaa-aa`. It works locally because local replica has only 1 subnet. With multiple subnets, replica won't know how to route `aaaaa-aa`. I think our original code is correct, if it doesn't work, it's probably a bug in the bitcoin canister.

Reply:
The canister IDs of Mainnet/Testnet Bitcoin Integration Canisters are considered to be implementation detail which is not a part of the specification. Having to hardcoding those canister IDs in agents is logically wrong.

IMO, the spec is correct. And the replica (or maybe the boundary_node) should be fixed to handle this case properly.

Reply:
Without the fix, the query call failed on both local replica and the mainnet.
After the fix, the query call succeed on local replica, but failed on the mainnet.